app/src/main/java/com/strv/movies/ui/moviedetail/MovieDetailScreen.kt

```diff
@@ -22,21 +23,42 @@ import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.viewmodel.compose.viewModel
 import coil.compose.AsyncImage
 import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.YouTubePlayer
 import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.AbstractYouTubePlayerListener
 import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView
 import com.strv.movies.R
 import com.strv.movies.data.OfflineMoviesProvider
 import com.strv.movies.model.MovieDetail
+import com.strv.movies.ui.error.ErrorScreen
+import com.strv.movies.ui.loading.LoadingScreen
+
+@Composable
+fun MovieDetailScreen(
+    viewModel: MovieDetailViewModel = viewModel()
+) {
+    val viewState by viewModel.viewState.collectAsState()
+
+    if (viewState.loading) {
+        LoadingScreen()
+    } else if (viewState.error != null) {
+        ErrorScreen(errorMessage = viewState.error!!)
+    } else {
+        viewState.movie?.let { MovieDetail(movie = it) }
+    }
+}
 
-@SuppressLint("UnrememberedMutableState")
 @Composable
 fun MovieDetail(movie: MovieDetail) {
     Column {
+        val (videoProgress, setVideoProgress) = rememberSaveable { mutableStateOf(0f) }
+        Log.d("TAG", "MovieDetail: $videoProgress")
```

Comment:
Not a good practice leaving logs in production code

Reply:
True, but for app for education it makes sense to keep it here and not bother with Timber or something.